core/templates/dev/head/pages/exploration_editor/ExplorationStatesService.js

```diff
@@ -109,6 +109,14 @@ oppia.factory('ExplorationStatesService', [
     var getStatePropertyMemento = function(stateName, backendName) {
       var accessorList = PROPERTY_REF_DATA[backendName];
       var propertyRef = _states.getState(stateName);
+      if (propertyRef === undefined) {
+        throw Error('Undefined states error debug logs:' +
```

Comment:
I think you can use $log.error. Try to test and see what the logs look like server-side (in your terminal).

Reply:
I take back what I said about $log.error, it looks like it does not actually log to the backend. So throw Error is better.

That said, "throw Error" halts execution completely. So here's what I would suggest instead (note that you'll still want to test the additionalInfo message manually and adjust as needed, it's going to do [Object object] in its current form):

```
      try {
        // put the code that might throw an error in here
      } catch(e) {
        var additionalInfo = (
          '\nUndefined states error debug logs:' +
          '\nRequested state name: ' + stateName +
          '\nRequested info type: ' + backendName +
          '\nExploration ID: ' + ContextService.getExplorationId() +
          '\nChange list: ' + ChangeListService.getChangeList() +
          '\nAll states names: ' + _states.getStateNames());
        e.message += additionalInfo;
        throw e;
      }
```

This ensures that the original error shows up too.